client/src/app/components/Loginform.js

```diff
@@ -22,20 +23,42 @@ export  class Loginform extends React.Component{
           data:{...this.state.data,[e.target.name]:e.target.value}
       
       })
-    onSubmit=e => {
-      e.preventDefault();
-  
-    const errors =this.validate(this.state.data);
-    this.setState({errors})
-    if(Object.keys(errors).length ===0){
-        this.props.submit(this.state.data);
-    }
-  
-  };
-  
+      onSubmit=(e) => {
+        e.preventDefault();
+        
+          this.setState({ errors:{}, loading:true});
+          
+          this.props.submit(this.state.data)
```

Comment:
if we have `props` you should pass them to the constructor right?

Reply:
done